### PR TITLE
Feature/sound rescan

### PR DESCRIPTION
**Pull request by mlightning3**
This gives SoundChan the ability to have sounds added/removed as long as they are in the top level directory as specified in soundchan.properties (recursive rescan to come). 

Also watches over usersounds.properties so changes in that are also applied.

### CODE REVIEW

File: src/main/java/soundchan/BotListener/BotListener.java [hunk 1]
```diff
@@ -59,14 +64,27 @@ private void loadProperties(@NotNull Properties properties) {
         if(audioOnUserJoin) {
             String userAudioPath = properties.getProperty("userAudioFilePath");
             if(userAudioPath == null || userAudioPath.contentEquals("")) {
-                localManager = new LocalAudioManager(localFilePath, "usersound.properties");
+                userAudioPath = "usersound.properties";
             }
-            else {
-                localManager = new LocalAudioManager(localFilePath, userAudioPath);
+            localManager = new LocalAudioManager(localFilePath, userAudioPath);
+
```
Comment (bwaggone) [inline src/main/java/soundchan/BotListener/BotListener.java:69]: I see these two blocks are almost identical, probably could make it into a function with an optional argument

Comment (bwaggone) [reply src/main/java/soundchan/BotListener/BotListener.java:69]: That would be the blocks on line 71 and 81

File: src/main/java/soundchan/BotListener/DirectoryWatcher.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            System.out.println("Error setting up watcher for " + filepath);
+        }
+    }
+
+    /**
+     * Called by an executor, checks for changes in the directory
+     */
+    public void run() {
+        try {
+            while(true) {
```
Comment (bwaggone) [inline src/main/java/soundchan/BotListener/DirectoryWatcher.java]: Add some sort of sleep here, we don't need it polling as fast as possible, maybe once every 5-10 seconds

File: src/main/java/soundchan/BotListener/UserSoundWatcher.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    private Path soundFileDirectory;
+    private WatchService watchService;
+    private WatchKey watchKey;
+
+    @SuppressWarnings("unchecked")
+    static <T> WatchEvent<T> cast(WatchEvent<?> event) {
+        return (WatchEvent<T>) event;
+    }
+
+    public UserSoundWatcher(LocalAudioManager audioManager, String filepath) {
```
Comment (bwaggone) [inline src/main/java/soundchan/BotListener/UserSoundWatcher.java]: For simplicity, try and combine this class and Directory watcher, they are almost identical after all. You could subclass them off of the same base class (defined by you), but I'm not so sure they need to be in separate files

**Comment by mlightning3** [comment]
Merged the Directory and File watcher classes and made function to add added watches to pool of worker threads. Also can define how long you want watchers to sleep too if you want.

**Comment by mlightning3** [comment]
Even though SoundChan doesn't build a list of sounds from sub directories currently, it can monitor changes to files in sub directories now.

**Review by bwaggone — APPROVED** [review]

**Pull request merged**